azure-keyvault/azure/keyvault/keys/_client.py

```diff
@@ -58,236 +44,552 @@ def create_config(**kwargs):
         config.retry = RetryPolicy(**kwargs)
         config.redirect = RedirectPolicy(**kwargs)
 
-        # TODO: these are requests-specific
-        config.cert = config.timeout = None
-        config.verify = True
         return config
 
-    def __init__(self, vault_url, credentials, config=None, transport=None):
-        # type: (str, Any, Configuration, HttpTransport) -> None
+    def __init__(self, vault_url, credentials, config=None, transport=None, **kwargs):
+        # type: (str, Any, Configuration, HttpTransport, Mapping[str, Any]) -> None
+        # TODO: update type hint for credentials
+        if not credentials:
+            raise ValueError("credentials")
+
+        if not vault_url:
+            raise ValueError("vault_url")
+
         self.vault_url = vault_url.strip("/")
-        config = config or KeyClient.create_config()
-        transport = RequestsTransport(config)
+        config = config or KeyClient.create_config(**kwargs)
+        transport = RequestsTransport(config.connection)
         policies = [
             config.user_agent,
             config.headers,
-            BearerTokenCredentialPolicy(credentials),
+            _BearerTokenCredentialPolicy(credentials),
             config.redirect,
             config.retry,
             config.logging,
         ]
         self._pipeline = Pipeline(transport, policies=policies)
-        models = {
-            "DeletedKey": DeletedKey,
-            "DeletedKeyItem": DeletedKeyItem,
-            "DeletedKeyItemPaged": DeletedKeyItemPaged,
-            "JsonWebKey": JsonWebKey,
-            "Key": Key,
-            "KeyAttributes": KeyAttributes,
-            "KeyCreateParameters": KeyCreateParameters,
-            "KeyItem": KeyItem,
-            "KeyItemPaged": KeyItemPaged,
-        }
-        self._deserialize = Deserializer(models)
-        self._serialize = Serializer(models)
 
     def create_key(
         self,
```

Comment:
in other languages we've split this into create_rsa_key and create_ec_key.  This is because size is only applicable to RSA, and curve is only applicable to EC.  Also then instead of key_type you only need a parameter specifying if it's HSM protected, which matches up better with the signature of import_key

Reply:
Is there a canonical example of what the surface should look like? Should we wait for the review?